cyences_app_for_splunk/bin/sophos_instance_details.py

```diff
@@ -0,0 +1,217 @@
+                    }
+                    response = requests.get(SOPHOS_TENANT_DICT[tenant]+"/endpoint/v1/endpoints?hostnameContains="+hostname+"ipAddresses="+ip, headers=requestHeaders)
+                    response_json = response.json()
+                    if response.status_code == 200:
+                        for instance in response_json.get("items"):
+                            list_of_uuid.append(instance)
+                    else:
+                        self.logger.error("Error while fetching UUID of instance")
+
+            elif(ip!="" and ip!=" "):
+                for tenant in SOPHOS_TENANT_DICT:
+                    requestHeaders = {
+                        "X-Tenant-ID": tenant,
+                        "Authorization": "Bearer "+ barier_token,
+                        "Accept": "application/json"
+                    }
+                    response = requests.get(SOPHOS_TENANT_DICT[tenant]+"/endpoint/v1/endpoints?ipAddresses="+ip, headers=requestHeaders)
+                    response_json = response.json()
+                    if response.status_code == 200:
+                        for instance in response_json.get("items"):
+                            list_of_uuid.append(instance)
+                    else:
+                        self.logger.error("Error while fetching UUID of instance")
+
+
+            elif(hostname!="" and hostname!=" "):
+                for tenant in SOPHOS_TENANT_DICT:
+                    requestHeaders = {
+                        "X-Tenant-ID": tenant,
+                        "Authorization": "Bearer "+ barier_token,
+                        "Accept": "application/json"
+                    }
+                    response = requests.get(SOPHOS_TENANT_DICT[tenant]+"/endpoint/v1/endpoints?hostnameContains="+hostname, headers=requestHeaders)
+                    response_json = response.json()
+                    if response.status_code == 200:
+                        for instance in response_json.get("items"):
+                            list_of_uuid.append(instance)
+                    else:
+                        self.logger.error("Error while fetching UUID of instance")
+
+            return list_of_uuid
+        except Exception as e:
+            self.logger.error("Error while fetching Instance UUID : {}".format(e))
+
+
+
+    def generate(self):
+
+        ip = self.ip
+        hostname = self.hostname
+        uuid = self.uuid
+        all_endpoints = self.all_endpoints
+        if(ip=="" and hostname=="" and uuid=="" and ip==" " and hostname==" " and uuid==" " and all_endpoints==False):
+            self.logger.error("Please provide IP or Hostname or UUID")
+            exit()
+
+        # Read Client ID & Client Secret
+        client_id,client_secret = self.get_client_details()
+
+        if not client_id or not client_secret:
+            self.logger.error("Sophos Client ID or Client Secret or both not found in the cs_configurations.conf file.")
```

Comment:
add line:
raise Exception("Sophos Client ID or Client Secret or both not found in the cs_configurations.conf file.")